.github/workflows/release-trigger.yaml

```diff
@@ -0,0 +1,26 @@
+# This Github action is the release trigger for Azure/karpenter-provider-azure
+# It will auto-generate the next semantic release version and tag the git ref
+
+name: Release Trigger
+on:
+    workflow_dispatch:
+
+jobs:
+  generate-sem-ver:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+        with:
+          egress-policy: audit
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
```

Comment:
This will need to be pinned, but can let dependabot take care of this

Reply:
It needs to be pinned to let `dependabot` take care of it? Pinning to the major was the given example in the docs, so was following that.

Reply:
[Pinned dependencies](https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies) is best practice from security perspective. Now that I think about it, dependabot won't switch unpinned to pinned (I think), so may need to do this manually once.

Reply:
Pinned:
https://github.com/actions/setup-node/commit/60edb5dd545a775178f52524783378180af0d1f8

Reply:
Pinned:
https://github.com/actions/setup-node/commit/60edb5dd545a775178f52524783378180af0d1f8

Reply:
Pinned:
https://github.com/actions/setup-node/commit/60edb5dd545a775178f52524783378180af0d1f8